### ISSUE TITLE
User-unfriendly structure of files in root folder

### ISSUE THREAD

**Issue by Masterjun3** [TASVideos/BizHawk#2654]
There are several issues with the main folder of default BizHawk. On their own they are just minor nitpicks, but together these issues create unnecessary complications that could be easily avoided.

- Main executable **is not** called `BizHawk.exe`, but instead `EmuHawk.exe`.
- Barely any mention of the term "EmuHawk" anywhere, so people aren't sure if opening `EmuHawk.exe` is correct.
- Multiple .exe files in the main folder. This causes even more confusion, because the tool is not installed with an installer but must be manually extracted.
- .config files are not known by default on Windows, so this is really only minor, but Windows hides file extensions by default, so the folder will appear to have files: `DiscoHawk`, `DiscoHawk.exe` (the .config file), `EmuHawk`, and `EmuHawk.exe` (the .config file).

All these together cause uncertainty when users run into issues starting the tools. It also adds unnecessary troubleshooting steps for those trying to help.

The problems could be easily mitigated, for example, by rename the main executable, moving DiscoHawk into a separate folder, and possibly renaming the .config file completely or at least remove the .exe from the filename.

**Comment by nattthebear** [comment]
I don't believe the `.config` can be easily changed.  That's a low level .NET infrastructure file, and they always have the name `{imageName}.config` and are opened automatically by the system and used to drive some low level decisions before our code is ever run.

It's funny because your assessment of the situation is right, so this issue presumably applies to a huge number of applications that all use .NET `.config` infra.  That's the way it is, I suppose.  Hide file extensions for known file types was one of the stupidest decisions Microsoft ever made.

**Comment by zeromus** [comment]
You've actually helped someone who said "help I don't know what program to run"? I've helped a lot of stupid people but I can't remembering hearing ANY of these as issues yet, although come to think of it the .config thing should have thrown some people for a loop.

**Comment by Masterjun3** [comment]
> You've actually helped someone who said "help I don't know what program to run"?

Yes this happens.

![image](https://user-images.githubusercontent.com/22375320/110172323-40113700-7dfd-11eb-8d2c-6830e1de1d78.png)

**Comment by zeromus** [comment]
Oh, I get it. I may have seen that, but it would have been invisible to me. You just ignore idiots like that and they eventually realize that it would take them less time to try both things than it would to ask which one thing to try and wait for someone to answer.

**Comment by YoshiRulz** [comment]
I agree with most of your points (in the Linux world, .NET Framework apps like ours are very unidiomatic) but:
> Barely any mention of the term "EmuHawk" anywhere, so people aren't sure if opening `EmuHawk.exe` is correct.

![Screenshot_20210306_064748](https://user-images.githubusercontent.com/13409956/110172034-4c6eb180-7e48-11eb-8376-4b8d640aedc5.png)

If these users you've run into can't be bothered to read the second section of readme, clearly labelled "Installing", it's not worth yours or anyone's time to spoonfeed them.

**Comment by Masterjun3** [comment]
Ah okay, sadly it's not usual for people to check out the source code repository page for a Readme.

Then maybe it would make sense to include a Readme in the package you download itself, so that people don't have to point to a dynamic repository?

**Comment by TiKevin83** [comment]
Unless it's somehow complicated to rename the executable from EmuHawk to BizHawk, I don't think "not catering to idiots" is a great reason to obfuscate the main executable's name

**Comment by adelikat** [comment]
It's EmuHawk on purpose.  EmuHawk is the emulator, DiscoHawk is the CD utility, Bizhawk is the overall suite of solutions

**Comment by Meerkov** [comment]
Allow me to contradict adlikat here with some arguments against using EmuHawk.

1. Name Recognition
Google search shows EmuHawk has <4k results for me.
Bizhawk has >200k results.
2. Inconsistent use by developers
BizHawk is advertised on this very github as such:
```
BizHawk is a multi-system emulator written in C#.
```
By adlikat's definition here, this is actually incorrect. BizHawk is not the emulator.
3. TasVideos users refer to the emulator as "BizHawk"
Let's take a look at one recent Moon run:
http://tasvideos.org/6980S.html
Emulator used: BizHawk
I don't know how to easily search how runs use one particular emulator over another, but we can do free-form text search, so let's do that next:
```
Search Results for Bizhawk on the site: 168 results http://tasvideos.org/Search.html?key=bizhawk
Search Results for EmuHawk on the site: 9 results (and most of them are documentatoin, not actual users) http://tasvideos.org/Search.html?key=emuhawk
```

Given all this evidence, I'm inclined to think that the only people who see a distinction between the brand name "BizHawk" and the emulator, is probably the dev team itself. I think renaming EmuHawk.exe as BizHawk.exe is a sensible change.

**Comment by YoshiRulz** [comment]
Oh hey, this thread.

---

Masterjun:
> Ah okay, sadly it's not usual for people to check out the source code repository page for a Readme. [...] maybe it would make sense to include a Readme in the package you download itself [...]

I do intend to include end-user documentation and the changelog in release artifacts. But then, if you're not capable of opening EmuHawk, a readme file won't help you...

---

TiKevin:
> Unless it's somehow complicated to rename the executable from EmuHawk to BizHawk [...]

In the Linux world, this is just not done—though in our case it wouldn't affect Linux users. For Windows users, I'm sure people would be upset if they had to update Batch scripts for no reason, or worse, adapt them to work with both. I don't think it would inconvenience us as developers any more than end-users.

> I don't think "not catering to idiots" is a great reason to obfuscate the main executable's name

lmao

---

Meerkov:
> 1) Name Recognition [...] Google search

Google hits are a proxy for popularity, and I don't think that necessarily gives rise to any renaming. People say "LEGOs" and "Googled" still, and neither have been adopted.

> 2) Inconsistent use by developers [... GitHub repo description]

This has been due a revamp for years.

> 3) TASVideos [submission metadata]

Was also due a revamp for years, then it got one, and now this field could easily be changed by anybody (personally, it's quite far down my list).

---

In summary, the software must serve the end-user, but this is a question of naming (as in *trademarks*) so the userbase doesn't get to decide.

I think that's everything covered, and I don't see anything actionable left. I'll leave this open until releases include some sort of end-user docs.

**Comment by Masterjun3** [comment]
> I think that's everything covered

What about my suggestion to move DiscoHawk.exe into a subfolder? I haven't seen a single person address this.
If there is only one .exe in the main folder, more people will figure it out by process of elimination. If there is no other .exe to start, it's likely to be the correct one, especially if it's the only one with an proper file icon.
